src/main/webapp/calendar-script.js

```diff
@@ -43,8 +66,125 @@ function getEvents(calendar) {
+  addressLine.appendChild(document.createElement('br'));
+  infoDisplay.appendChild(addressLine);
+
+  // infoDisplay.appendChild(address);
+  modalBody.appendChild(infoDisplay);
+
+  // create new div to hold map
+  const mapDis = document.createElement('div');
+  mapDis.id = 'map';
+  modalBody.appendChild(mapDis);
+
+  // instantiate map
+  const map = new google.maps.Map(document.getElementById('map'), {
+    zoom: zoomThirteen
+  });
+
+  const service = new google.maps.places.PlacesService(map);
+  service.getDetails({
+    placeId: eventObj.extendedProps.placeId
+  }, function(result, status) {
+    if (status != google.maps.places.PlacesServiceStatus.OK) {
+      alert(status);
+      return;
+    }
+    // set center of map
+    map.setCenter(result.geometry.location);
+  
+    // add open hours to mod info Display
+    let openHours = document.createElement('p');
+
+    // if there are open hours display them, otherwise open all day
+    try {
+      const stringFullDate = eventObj.extendedProps.stringDate;
+      const dateStr = stringFullDate.split('T')[0]; 
+      const intOfWeek = getIntOfWeek(dateStr);
+
+      // create bold open and time element
+      const open = document.createElement('p');
+      const openTime = formatAMPM(result.opening_hours.periods[intOfWeek].open.time); 
+      const boldOpen = document.createElement('b');
+      boldOpen.innerText = 'Open: ';
+      open.appendChild(boldOpen);
+      open.innerHTML += openTime;
+      open.appendChild(document.createElement('br'));
+      openHours.appendChild(open);
+
+      // create bold close and time element
+      const close = document.createElement('p');
+      const closeTime = formatAMPM(result.opening_hours.periods[intOfWeek].close.time);
+      const boldClose = document.createElement('b');
+      boldClose.innerText = 'Close: ';
+      close.appendChild(boldClose);
+      close.innerHTML += closeTime;
+      close.appendChild(document.createElement('br'));
+      openHours.appendChild(close);
+    } catch(e) {
+      // create bold element open all day if no hours available
+      const boldAllDay = document.createElement('b');
+      boldAllDay.innerText = 'Open All Day ';
+      openHours.appendChild(boldAllDay);
+    }
+
+    infoDisplay.appendChild(openHours);
+
+    const marker = new google.maps.Marker({
+      map: map,
+      position: result.geometry.location
+    });
+  });
+}
+
+// takes a string in date format and return the int of the week
+function getIntOfWeek(date) {
+  const intOfWeek = new Date(date).getDay();    
+  return isNaN(intOfWeek) ? null : 
+    intOfWeek;
+}
+
+// converts from 24hr (hh:mm) to 12hr format (hh:mm AM/PM)
+function formatAMPM(time) {
+  let hours = time.substring(0, 2);
+  let minutes = time.substring(2, 4);
+  const ampm = hours >= 12 ? 'pm' : 'am';
+  hours = hours % 12;
+  hours = hours ? hours : 12; // the hour '0' should be '12'
+  minutes = (minutes < 10 && minutes > 0) ? '0' + minutes : minutes;
```

Comment:
why is this conversion necessary? if you're going from 24hr hh:mm to 12hr hh:mm,  the substring you get on line 180 should already have the '0' in front, no?

Reply:
Yes you are right! I was thinking in Java where parsing "03" from a string converts it to 3